pkg/ocm/ocm_test.go

```diff
@@ -0,0 +1,147 @@
+package ocm_test
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	awsv1alpha1 "github.com/openshift/aws-account-operator/pkg/apis/aws/v1alpha1"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"k8s.io/apimachinery/pkg/util/json"
+
+	"github.com/openshift/configuration-anomaly-detection/pkg/ocm"
+	mocks "github.com/openshift/configuration-anomaly-detection/pkg/ocm/mock"
+)
+
+var _ = Describe("OCM", func() {
+	var (
+		mockCtrl          *gomock.Controller
+		client            *ocm.OcmClient
+		mocOCMConnection  *mocks.MockocmHandlerIf
+		clusterDeployment *hivev1.ClusterDeployment
+		awsAccountClaim   *awsv1alpha1.AccountClaim
+		err               error
+		clustername       string
+		supportRoleArn    string
+	)
+	Describe("When fetching a ClusterDeployment", func() {
+		BeforeEach(func() {
+			mockCtrl = gomock.NewController(GinkgoT())
+			mocOCMConnection = mocks.NewMockocmHandlerIf(mockCtrl)
+			client = &ocm.OcmClient{Comm: mocOCMConnection}
```

Comment:
so this does seem to be worse.
you can have a `func New() ocm.OcmClient` that will take the connection / take an options struct that holds the connection struct

Reply:
makes sense

Reply:
@georgettica  https://github.com/openshift/configuration-anomaly-detection/pull/26/commits/0277c38a13c8fce6100ff346f53dc2e9a10b3e5f

Is this what you had in your mind? I introduced a constructor, but I cannot make the OcmClient struct private, because we are running the tests in a separate package called `ocm_test`.